asmcomp/i386/proc.ml

```diff
@@ -107,7 +107,7 @@ let word_addressed = false
    These globals are denoted by Incoming and Outgoing stack locations
    with negative offsets, starting at -64.
    Unlike arguments passed on stack, arguments passed in globals
-   do not prevent tail-call elimination.  The caller stores arguments
+   do not prevent tailcall elimination.  The caller stores arguments
```

Comment:
To answer the last remark, the `check-typo` script is already quite slow and its purpose is to check low-level conventions on the code. You should probably add a new script for spelling errors.